src/codeu/chat/server/Controller.java

```diff
@@ -64,9 +94,21 @@ public Message newMessage(Uuid id, Uuid author, Uuid conversation, String body,
 
     if (foundUser != null && foundConversation != null && isIdFree(id)) {
 
-      message = new Message(id, Uuid.NULL, Uuid.NULL, creationTime, author, body);
+      message = new Message(id, Uuid.NULL, Uuid.NULL, creationTime, author, body,conversation);
       model.add(message);
-      LOG.info("Message added: %s", message.id);
+      if(localFile != null && isInitialized)
```

Comment:
You should not allow `localFile` to be null, because if it is null, this class will not be in a working state. You can guarantee that it's never null by deleting the constructor that doesn't include it (the one that looks like `public Controller(Uuid serverId, Model model)`)